src/ipa_pc/mod.rs

```diff
@@ -828,36 +1150,37 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
 
         // h(X) polynomial added to the set of polynomials for multi-poly single-point batching
         let mut labeled_polynomials = labeled_polynomials;
+        let batch_degree = batch_polynomial.degree();
         let labeled_batch_polynomial = LabeledPolynomial::new(
             format!("Batch"),
             batch_polynomial,
             None,
-            None
+            Some(batch_degree)
```

Comment:
There is a mistake here, and I realized it having unit tests failing in Marlin:

```suggestion
            if has_hiding { Some(batch_degree) } else { None }
```

Please change and try to understand why the UTs in poly-commit didn't capture it

Reply:
This fixed, thank you. Regarding UT - in poly-commit the UTs are built so they always has hiding bound, this is why this issue wasn't catched. You can check get_data_for_verifier function, hiding_bound is always "something" there

Reply:
Can we modify the tests to test also hiding_bound = false ? I guess arkworks guys never did it cause applications that are using PC (like Marlin) are ZK by default

Reply:
Done